File: services/auth/src/repositories/userRepository.ts
import { User } from "../models";

const data: User[] = []

export class UserRepository {
  save(payload: User) {
    data.push(payload)
  }

  findById(id: string) { }

  findByEmail(email: string) {
    const user = data.find(u => u.email === email)

    if (!user) throw new Error()

    return user
  }
}

File: back/src/main/java/br/devhub/repositories/UserRepository.java
package br.devhub.repositories;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

import br.devhub.models.User;

public interface UserRepository extends JpaRepository<User, String> {
	@Query(nativeQuery = true, value = "SELECT * FROM gestao.pessoa WHERE email = :email")
	List<User> findByEmail(@Param("email") String email);
}
